### PR TITLE
Feature/unique module key

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: src/main/java/org/sonarqube/gradle/SonarQubePlugin.java [hunk 1]
```diff
@@ -339,14 +338,14 @@ private void configureJdkSourceAndTarget(Project project, Map<String, Object> pr
...
-    // the Gradle project to which the plugin is applied contains special characters.
-    // (':' works, ' ' doesn't.) In such a case, sonar.projectKey can be overridden manually.
-    String name = project.getName();
-    String group = project.getGroup().toString();
-    return group.isEmpty() ? name : (group + ":" + name);
+    Project rootProject = project.getRootProject();
+    String rootProjectName = rootProject.getName();
+    String rootGroup = rootProject.getGroup().toString();
+    String rootKey = rootGroup.isEmpty() ? rootProjectName : (rootGroup + ":" + rootProjectName);
+    if (project == rootProject) {
```
Comment (SonarTech) [inline src/main/java/org/sonarqube/gradle/SonarQubePlugin.java]: MAJOR Change this comparison to use the equals method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: src/main/java/org/sonarqube/gradle/SonarQubePlugin.java [hunk 2]
```diff
@@ -339,11 +338,7 @@ private void configureJdkSourceAndTarget(Project project, Map<String, Object> pr
...
   private static String getProjectKey(Project project) {
-    // SonarQube uses project keys in URL parameters without internally URL-encoding them.
-    // According to my manual tests with sonar-gradle plugin based on Sonar Runner 2.0 and Sonar 3.4.1,
-    // the current defaults will only cause a problem if project.group or project.name of
-    // the Gradle project to which the plugin is applied contains special characters.
-    // (':' works, ' ' doesn't.) In such a case, sonar.projectKey can be overridden manually.
+    String path = project.getPath();
     String name = project.getName();
     String group = project.getGroup().toString();
     return group.isEmpty() ? name : (group + ":" + name);
```
Comment (dbmeneses) [inline src/main/java/org/sonarqube/gradle/SonarQubePlugin.java]: is this correct? wasn't the new naming strategy involve the path?

Comment (dbmeneses) [inline src/main/java/org/sonarqube/gradle/SonarQubePlugin.java]: nevermind, it's only for root project. I would rename the method and remove the useless path var.

**Comment by dbmeneses** [comment]
LGTM. 
Ignore my previous comments on the first commit.

**Comment by henryju** [comment]
@dbmeneses Yep sorry I decided to commit rework of module ids as a dedicated commit but I forgot this "path" variable that is only used in the second commit.

**Pull request merged**